### PR TITLE
[MongoDB Parsing 2/3] Add the types header file for MongoDB Frames

### PR DESCRIPTION
**Pull request by kpattaswamy**
Summary: This PR adds the header file the parsing logic uses (currently in #1704 ) to get the MongoDB wire protocol's frame specifications. The header file currently contains the frame spec for a standard message header, the spec of a payload of type `OP_MSG` and contains the basic structure for a `Record`.

Related issues: https://github.com/pixie-io/pixie/issues/640

Type of change: /kind feature

Test Plan: Tested the integration of this and the parsing logic through unit tests.

### CODE REVIEW

File: src/stirling/source_connectors/socket_tracer/protocols/mongodb/types.h [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+#include "src/stirling/utils/utils.h"
+
+namespace px {
+namespace stirling {
+namespace protocols {
+namespace mongodb {
+
+enum class Type : int32_t {
+  kOPMsg = 2013,
+  kOPReply = 1,
```
Comment (ddelnano) [inline src/stirling/source_connectors/socket_tracer/protocols/mongodb/types.h:34]: While this isn't in our style guide, all of the existing enums and enum classes are sorted by value. Therefore, I think it's best to maintain that consistency and move this after `kOPCompressed = 2012,`.

File: src/stirling/source_connectors/socket_tracer/protocols/mongodb/types.h [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    return absl::Substitute("req=[$0] resp=[$1]", req.ToString(), resp.ToString());
+  }
+};
+
+struct ProtocolTraits : public BaseProtocolTraits<Record> {
+  using frame_type = Frame;
+  using record_type = Record;
+  using state_type = NoState;
+};
+
```
Comment (ddelnano) [inline src/stirling/source_connectors/socket_tracer/protocols/mongodb/types.h:144]: I think we should add the `src/stirling/source_connectors/socket_tracer/protocols/mongodb/BUILD.bazel` file changes to use this source file within this PR (cc_library / pl_cc_library).

**Review by ddelnano — COMMENTED** [review]
I didn't get a chance to familiarize myself with the mongo docs referenced, so I need another pass for this review. But I added a few minor comments for my first pass.

**Comment by ddelnano** [comment]
@kpattaswamy this lgtm. @etep can you give this a look as well?

**Review by etep — APPROVED** [review]

**Review by JamesMBartlett — APPROVED** [review]

**Pull request merged**